### PR TITLE
Support .kt file generation

### PR DESCRIPTION
**Pull request by hotchemi**
## Issues

- resolves https://github.com/hotchemi/PermissionsDispatcher/issues/320
- also resolves https://github.com/hotchemi/PermissionsDispatcher/issues/327

## Overview

- Add Kotlin file generation support.
  - Add KotlinPoet, `permissions.dispatcher.processor.impl.kotlin` classes to generate .kt file at compile time.
  - add `kotlin` method to `RuntimePermissions` to specify generate .kt file or not.
    - The reason of it is I couldn't find a way to detect file type from annotation processing info.
    - Plus, this feature is still kind of `experimental` so I'd like to allow user to generate .java file even targeted file is Kotlin.
- Address internal method
  - It seems we can address by 3c0e4fabe8a5bf386234ff45323abc292465a916...really? 😅
- Update `sample-kotlin` to use new API!

## Misc

- It turned out that it's sort of tough to support both Java/Kotlin than I expected.
  - But not sure how we can cope with it 🤔

### CODE REVIEW

File: processor/src/main/kotlin/permissions/dispatcher/processor/exception/SupportV13MissingException.kt [hunk 1]
```diff
@@ -1,5 +0,0 @@
-package permissions.dispatcher.processor.exception
```
Comment (hotchemi) [inline processor/src/main/kotlin/permissions/dispatcher/processor/exception/SupportV13MissingException.kt]: no longer used.

File: gradle.properties [hunk 2]
```diff
@@ -5,21 +5,23 @@ USER = hotchemi
 GROUP_ID = com.github.hotchemi
 ARTIFACT_ID_LIBRARY = permissionsdispatcher
 ARTIFACT_ID_PROCESSOR = permissionsdispatcher-processor
-VERSION = 2.5.0-SNAPSHOT
+VERSION = 3.0.0-SNAPSHOT
```
Comment (hotchemi) [inline gradle.properties]: 🗡

Comment (aurae) [inline gradle.properties]: WHOOO!

**Comment by hotchemi** [comment]
Not a regular work but I published 3.0.0-SNPSHOT to jfrog to prepare for today's talk at local meetup 🙇

**Comment by shiraji** [comment]
Good luck! 🤞

**Comment by hotchemi** [comment]
I suppose, we need to fix `test` and `test-v13` project to use generated .kt files right? 👀

**Comment by hotchemi** [comment]
About https://github.com/hotchemi/PermissionsDispatcher/pull/344#issuecomment-316297145, I'll create another PR! here we go!

**Comment by shiraji** [comment]
> I suppose, we need to fix test and test-v13 project to use generated .kt files right? 👀

As far as I know, yes!

**Pull request merged**